pwndbg/commands/kbase.py

```diff
@@ -6,48 +6,32 @@
 
 import pwndbg.color.message as M
 import pwndbg.commands
-import pwndbg.gdblib.memory
-import pwndbg.gdblib.vmmap
+import pwndbg.gdblib.kernel
 from pwndbg.commands import CommandCategory
 from pwndbg.gdblib.config import config
 
 parser = argparse.ArgumentParser(description="Finds the kernel virtual base address.")
 
+parser.add_argument("-r", "--rebase", action="store_true", help="rebase loaded symbol file")
+
 
 @pwndbg.commands.ArgparsedCommand(parser, category=CommandCategory.KERNEL)
 @pwndbg.commands.OnlyWhenQemuKernel
 @pwndbg.commands.OnlyWhenPagingEnabled
-def kbase() -> None:
+def kbase(rebase=False) -> None:
     if config.kernel_vmmap == "none":
         print(M.error("kbase does not work when kernel-vmmap is set to none"))
         return
 
-    arch_name = pwndbg.gdblib.arch.name
-    if arch_name == "x86-64":
-        # First opcode, seems to be consistent
-        magic = 0x48
-    elif arch_name == "aarch64":
-        # First byte of "MZ" header
-        magic = 0x4D
-    else:
-        print(M.error(f"kbase does not support the {arch_name} architecture"))
+    print(M.success(f"Found virtual text base address: {hex(pwndbg.gdblib.kernel.kbase())}"))
+
+    if not rebase:
         return
 
-    mappings = pwndbg.gdblib.vmmap.get()
-    for mapping in mappings:
-        # TODO: Check alignment
-        # TODO: Check if the supervisor bit is set for aarch64
-        if not mapping.execute:
-            continue
-        try:
-            b = pwndbg.gdblib.memory.byte(mapping.vaddr)
-        except gdb.MemoryError:
-            print(
-                M.error(
-                    f"Could not read memory at {mapping.vaddr:#x}. Kernel vmmap may be incorrect."
-                )
-            )
-            continue
-        if b == magic:
-            print(M.success(f"Found virtual base address: {mapping.vaddr:#x}"))
-            break
+    symbol_file = gdb.current_progspace().filename
+
+    if symbol_file:
+        gdb.execute("symbol-file", to_string=True)
```

Comment:
How does this work? Don't we need to pass in an argument?
```
pwndbg> help symbol-file
Load symbol table from executable file FILE.
Usage: symbol-file [-readnow | -readnever] [-o OFF] FILE
OFF is an optional offset which is added to each section address.
The `file' command can also load symbol tables, as well as setting the file
to execute.
The '-readnow' option will cause GDB to read the entire symbol file
immediately.  This makes the command slower, but may make future operations
faster.
The '-readnever' option will prevent GDB from reading the symbol file's
symbolic debug information.
```

Reply:
https://sourceware.org/gdb/current/onlinedocs/gdb.html/Files.html

> `symbol-file` with no arguments clears out GDB information on your program's symbol table.

Reply:
This will probably break some user's workflows in some edge cases, but it's fine for now (unless there's some way to find a loaded kernel symbol file and just clear that).

Reply:
Why do we need the `to_string=True`?